rust/connlib/tunnel/src/lib.rs

```diff
@@ -295,7 +290,7 @@ where
         Ok(())
     }
 
-    fn poll_sockets<'a>(&'a mut self, cx: &mut Context<'_>) -> Poll<device_channel::Packet<'a>> {
+    fn poll_sockets(&mut self, device: &mut Device, cx: &mut Context<'_>) -> Poll<io::Result<()>> {
```

Comment:
I was planning on just removing `ConnectionState` now that we don't have `Mutex`es

Reply:
Yeah I know. But I couldn't get this to work otherwise without allocating the list of received packets.

I think this will get easier once we shuffle around things.

Reply:
The problem is that I need access to the `Device` as part of looping over the packets. I think that should become easier as we flatten the number of components here.

Can we still merge this and just refactor after it?

Reply:
Yeah, #3738 will need something similar